ouroboros-consensus/src/Ouroboros/Consensus/Fragment/InFuture.hs

```diff
@@ -47,10 +45,9 @@ import           Ouroboros.Consensus.Util.Time
 
 data CheckInFuture m blk = CheckInFuture {
        -- | POSTCONDITION:
-       --
-       -- > checkInFuture vf >>= \(af, fut) ->
-       -- >   validatedFragment vf == af <=> null fut
-       checkInFuture :: ValidatedFragment (Header blk) (LedgerState blk)
+       -- > checkInFuture _ af >>= \(af', fut) ->
```

Comment:
Perhaps we could use this opportunity to add more comments to this function (especially the return type).